client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -394,19 +397,33 @@ export class PlanFeatures2023Grid extends Component<
 	}
 
 	renderTabletView() {
-		const { planProperties, hideFreePlan } = this.props;
-		const plansToShow = hideFreePlan
-			? [ TYPE_PERSONAL, TYPE_PREMIUM, TYPE_BUSINESS, TYPE_ECOMMERCE, TYPE_ENTERPRISE_GRID_WPCOM ]
-			: [
-					TYPE_FREE,
-					TYPE_PERSONAL,
-					TYPE_PREMIUM,
-					TYPE_BUSINESS,
-					TYPE_ECOMMERCE,
-					TYPE_ENTERPRISE_GRID_WPCOM,
-			  ];
-		const topRowPlans = plansToShow.slice( 0, 3 );
-		const bottomRowPlans = plansToShow.slice( 3, 6 );
+		const { planProperties, hideFreePlan, flowName } = this.props;
+
+		let plansToShow = [];
+		let indexOfLastPlanToShowOnTop = 3;
+		const indexOfLastPlanToShowOnBottom = 6;
```

Comment:
I find these a little hard to follow, I think they are also misleading e.g. 

- the semantics of `indexOfLastPlanToShowOnTop` is not correct as used for the `slice` method. `3` in the case of `[ TYPE_FREE, TYPE_PERSONAL, TYPE_PREMIUM, TYPE_BUSINESS, TYPE_ECOMMERCE, TYPE_ENTERPRISE_GRID_WPCOM ]` would be `TYPE_BUSINESS` (indices starts at `0`) and that's not shown on the top, but the bottom.
- `indexOfLastPlanToShowOnBottom` - there is no plan with index `6` in any of the plan combinations.

Let's maybe rework these with a variable of the kind "number of items to show on the top", and figure out the second value (`indexOfLastPlanToShowOnBottom`) dynamically from the array length maybe

Reply:
Totally right! Changed it to reflect your suggestion 👍